src/main/java/servlets/SearchServlet.java

```diff
@@ -47,20 +40,20 @@
 public class SearchServlet extends HttpServlet {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String desiredLabel = request.getParameter("label");
-    ImmutableList<Receipt> receipts = getReceiptsWithMatchingLabel(desiredLabel);
+    String categories = request.getParameter("categories");
+    ImmutableList<Receipt> receipts = getReceiptsWithMatchingCategories(categories);
 
     Gson gson = new Gson();
     response.setContentType("application/json;");
     response.getWriter().println(gson.toJson(receipts));
   }
 
-  /** Returns ImmutableList of receipts from datastore with the same label as desiredLabel. */
-  private ImmutableList<Receipt> getReceiptsWithMatchingLabel(String desiredLabel) {
-    // Set filter to retrieve only receipts with label equal to desiredLabel.
-    Filter matchingLabels = new FilterPredicate("label", FilterOperator.EQUAL, desiredLabel);
+  /** Returns ImmutableList of receipts from datastore with the same categories. */
+  private ImmutableList<Receipt> getReceiptsWithMatchingCategories(String categories) {
```

Comment:
If only one category can be searched for this version, can you change all the plural "categories" to be singular.